src/video_mode/vga_320x200.s

```diff
@@ -19,7 +19,7 @@ vga_map_frame_buffer_loop:
     mov [_p1 + ecx * 8], eax
 
     add eax, 4096
-    cmp eax, 0xa0000 + 320 * 200
+    cmp eax, 0xbffff
```

Comment:
I think this should be an exclusive upper bound of `0xc0000` because we're using a `<` (not `<=`) operation through the `jl` instruction.

```suggestion
    cmp eax, 0xc0000
```

(Technically, this doesn't really matter since we increase eax in steps of 4096, but we still should do it correctly to avoid confusion.)